cmd/cloud.go

```diff
@@ -343,20 +344,69 @@ func getCmdCloud(gs *state.GlobalState) *cobra.Command {
 	}
 
 	exampleText := getExampleText(gs, `
-  {{.}} cloud script.js`[1:])
+  # Authenticate with Grafana k6 Cloud
+  $ {{.}} cloud login
+
+  # Run a k6 script in the Grafana k6 cloud
+  $ {{.}} cloud run script.js
+
+  # Run a k6 archive in the Grafana k6 cloud
+  $ {{.}} cloud run archive.tar
+
+  # [deprecated] Run a k6 script in the Grafana k6 cloud
+  $ {{.}} cloud script.js
+
+  # [deprecated] Run a k6 archive in the Grafana k6 cloud
+  $ {{.}} cloud archive.tar`[1:])
 
 	cloudCmd := &cobra.Command{
 		Use:   "cloud",
 		Short: "Run a test on the cloud",
-		Long: `Run a test on the cloud.
+		Long: `[deprecation notice]
+The k6 team is in the process of modifying and deprecating the cloud command behavior. In the future, the "cloud"
+command will only display a help text, instead of running tests in the cloud.
+To run tests in the cloud, users are now invited to migrate to the "k6 cloud run" command instead.
+
+Run a test on the cloud.
 
 This will execute the test on the k6 cloud service. Use "k6 login cloud" to authenticate.`,
-		Example: exampleText,
-		Args:    exactArgsWithMsg(1, "arg should either be \"-\", if reading script from stdin, or a path to a script file"),
+		Args:    exactCloudArgs(),
 		PreRunE: c.preRun,
 		RunE:    c.run,
+		Example: exampleText,
 	}
+
+	// Register `k6 cloud` subcommands
+	cloudCmd.AddCommand(getCmdCloudRun(gs))
+	cloudCmd.AddCommand(getCmdCloudLogin(gs))
+
 	cloudCmd.Flags().SortFlags = false
 	cloudCmd.Flags().AddFlagSet(c.flagSet())
+
 	return cloudCmd
 }
+
+func exactCloudArgs() cobra.PositionalArgs {
+	return func(_ *cobra.Command, args []string) error {
+		if len(args) < 1 || len(args) > 2 {
+			return fmt.Errorf("accepts 1 or 2 arg(s), received %d", len(args))
+		}
+
+		var (
+			isRunSubcommand   = args[0] == "run"
+			isLoginSubcommand = args[0] == "login"
+			isScript          = filepath.Ext(args[0]) == ".js"
+			isArchive         = filepath.Ext(args[0]) == ".tar"
```

Comment:
Do we really need such differentiation? 
Even further, do we really want this limitation (up to `.js` and `.tar`)? 

For instance, what if the user wants to run a test with the new `experimental_enhanced` compatibility mode, and submit a TS file?

Reply:
What about inverting the logic, and instead of looking for script/archive directly, based our check on whether the argument is **not** a subcommand.

Reply:
I've excluded this check 👍🏻